ddtrace/propagation/http.py

```diff
@@ -543,20 +543,24 @@ def my_controller(url, headers):
 
         try:
             normalized_headers = {name.lower(): v for name, v in headers.items()}
-            # Check all styles until we find the first valid match
-            # DEV: We want to check them in this specific priority order
-            if PROPAGATION_STYLE_DATADOG in config._propagation_style_extract:
-                context = _DatadogMultiHeader._extract(normalized_headers)
-                if context is not None:
-                    return context
-            if PROPAGATION_STYLE_B3 in config._propagation_style_extract:
-                context = _B3MultiHeader._extract(normalized_headers)
-                if context is not None:
-                    return context
-            if PROPAGATION_STYLE_B3_SINGLE_HEADER in config._propagation_style_extract:
-                context = _B3SingleHeader._extract(normalized_headers)
-                if context is not None:
-                    return context
+
+            # loop through the extract propagation styles specified in order
+            for prop_style in config._propagation_style_extract:
+                if prop_style == PROPAGATION_STYLE_DATADOG:
+                    context = _DatadogMultiHeader._extract(normalized_headers)
+                    if context is not None:
+                        return context
+
+                if prop_style == PROPAGATION_STYLE_B3:
+                    context = _B3MultiHeader._extract(normalized_headers)
+                    if context is not None:
+                        return context
+
+                if prop_style == PROPAGATION_STYLE_B3_SINGLE_HEADER:
+                    context = _B3SingleHeader._extract(normalized_headers)
+                    if context is not None:
+                        return context
```

Comment:
we could probably simplify this code block a bit and maybe make `prop_styles_to_propagator` a global dict?
```suggestion
           prop_styles_to_propagator = {PROPAGATION_STYLE_DATADOG: _DatadogMultiHeader, PROPAGATION_STYLE_B3:  _B3MultiHeader, PROPAGATION_STYLE_B3_SINGLE_HEADER: _B3SingleHeader}
           for prop_style in PROPAGATION_STYLE_DATADOG:
                if prop_style in config._propagation_style_extract:
                    propagator = prop_styles_to_propagator[prop_style]
                    context = propagator._extract(normalized_headers)
                    if context is not None:
                        return context
```